src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -77,39 +145,247 @@ public PolicyEntity updatePolicy(@PathParam("domainid") Long domainid, @PathPara
      */
     @Path("/{domainid}/policies/{policyid}")
     @DELETE
-    public PolicyEntity deletePolicy(@PathParam("domainid") Long domainid, @PathParam("policyid") Long policyid,
+    public void deletePolicy(@PathParam("domainid") Long domainid, @PathParam("policyid") Long policyid,
             PolicyEntity entity) {
 
-        logger.info("Deleting Policies Entity ...");
-        //TODO Sudhir: Add db calls here
-        return null;
+        logger.info("Deleting for Policies Entity Id...:" + policyid);
+
+        this.txControl.required(new Callable<Void>() {
+
+            @Override
+            public Void call() throws Exception {
+
+                DomainEntity result = em.find(DomainEntity.class, domainid);
+                if (result == null) {
+                    return null;
+                }
+                PolicyEntity policyresult = em.find(PolicyEntity.class, policyid);
+                if (policyresult == null) {
+                    return null;
+                }
+                em.remove(policyresult);
+                return null;
+            }
+        });
     }
 
     /**
      * Lists the Policy Id's for a given domain
      *
      * @return - Policy Id's
      */
-    @Path("/{domainid}")
+    @Path("/{domainid}/policies")
     @GET
     public List<String> getPolicyIds(@PathParam("domainid") Long domainid) {
 
-        logger.info("Listing Policy Ids'");
-	    //TODO Sudhir: Add db calls here
-		return null;
+        logger.info("Listing Policy Ids'for domain Id ...:" + domainid);
+
+
+        return this.txControl.supports(new Callable<List<String>>() {
+
+            @Override
+            public List<String> call() throws Exception {
+
+                DomainEntity result = em.find(DomainEntity.class, domainid);
+                if (result == null) {
+                    throw new Exception("Domain Entity does not exists...");
+                    //TODO - to add RETURN 404 error:Sudhir
+                }
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+                CriteriaQuery<PolicyEntity> query = criteriaBuilder.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = query.from(PolicyEntity.class);
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("domain").get("id"), domainid)));
+                List<PolicyEntity> policyList = em.createQuery(query).getResultList();
+
+                if (policyList.isEmpty()) {
+                    return null;
+                }
+                List<String> policy = new ArrayList<String>();
```

Comment:
policy -> policies, and right below this line you can do if (policyList.isEmpty()) {return policies}

Reply:
ok